wen-restart/src/last_voted_fork_slots_aggregate.rs

```diff
@@ -120,12 +133,22 @@ impl LastVotedForkSlotsAggregate {
         if from == &self.my_pubkey {
             return None;
         }
-        // If active peers didn't change, we don't need to update active stake in epoch info.
-        if self.active_peers.insert(*from) {
-            self.update_epoch_info_active_stake();
-        }
         let root_slot = self.root_bank.slot();
         let new_slots_vec = new_slots.to_slots(root_slot);
+        if new_slots_vec.is_empty() {
+            return None;
+        }
+        let last_vote_epoch = self
+            .root_bank
+            .get_epoch_and_slot_index(*new_slots_vec.last().unwrap())
+            .0;
+        if self
+            .node_to_last_vote_epoch_map
+            .insert(*from, last_vote_epoch)
+            != Some(last_vote_epoch)
+        {
+            self.update_epoch_info();
```

Comment:
do we need to update the entire map here, can we just update the entry for the `from` validator

Reply:
We don't. But right now when we allow updates, just updating the entry for the from validator would be a bit messy, because you may need to subtract stake from some epoc_info.

Can I clear up the update logic in the following PR forbidding entry update?

Reply:
Hmm is it more complicated than removing the validator's stake from the latest epoch_info they last voted for and adding to the new one??

Reply:
oh i see what you mean, when we prevent updates none of this matters

Reply:
Nah it's fine, I implemented it already. But it's going away.